mlir/lib/Conversion/VectorToXeGPU/VectorToXeGPU.cpp

```diff
@@ -97,6 +97,17 @@ static LogicalResult transferPreconditions(PatternRewriter &rewriter,
   return success();
 }
 
+// Common preconditions for the lowering of vector.gather and vector.scatter:
+//  1. Source is a memref.
+static LogicalResult gatherScatterPreconditions(PatternRewriter &rewriter,
```

Comment:
Could you also add negative test cases that cover preconditions?

Just a simple test to ensure matcher rejects what it's supposed to.
For example, see `@negative_gemm_transpose_a` in `contract-to-xegpu.mlir`.

Reply:
there's already a test called ['no_load_tensor'](https://github.com/llvm/llvm-project/pull/158024/files#diff-626169fad27915d9ac2cf273b1d647643268ef042854043f8cdfbb00be4e8d53R141), or do you mean something else?

Reply:
Perfect👍Missed it